packages/website/components/account/paymentTable.js/paymentTable.js

```diff
@@ -1,14 +1,32 @@
+import { useMemo } from 'react';
+
 import Tooltip from '../../../modules/zero/components/tooltip/tooltip.js';
 import InfoIcon from '../../../assets/icons/info';
 import Button from '../../button/button.js';
+import { formatAsStorageAmount, formatCurrency } from '../../../lib/utils.js';
+
+const getAdditionalStoragePrice = price => (typeof price === 'number' ? `${formatCurrency(price / 100)} / GiB` : 'N/A');
+
+const PaymentTable = ({ plans: plansProp, currentPlan, setPlanSelection, setIsPaymentPlanModalOpen }) => {
+  const plans = useMemo(() => {
+    const isCurrentPlanStandard = ['free', 'lite', 'pro'].includes(currentPlan?.id);
+
+    if (!isCurrentPlanStandard) {
+      return [currentPlan, ...plansProp.slice(1)];
+    }
+
+    return plansProp;
+  }, [plansProp, currentPlan]);
 
-const PaymentTable = ({ plans, currentPlan, setPlanSelection, setIsPaymentPlanModalOpen }) => {
   return (
     <>
       {currentPlan && (
         <p className="billing-content-intro" data-testid="currentPlanIndicator">
           <span>
-            Your current plan is: <strong data-testid="currentPlan.title">{currentPlan.title}</strong>
+            Your current plan is:{' '}
+            <strong data-testid="currentPlan.metadata['UI Label']">
+              {currentPlan?.metadata['UI Label'] ?? 'Custom'}
```

Comment:
Can you avoid adding metadata-dependent view logic in the jsx here by first parsing the thing-with-metadata to something else with explicit typing so you dont need to put defaults like 'Custom' in here?